src/app/modules/data-set/dialogs/add-project-dialog/add-project-dialog.component.ts

```diff
@@ -22,6 +25,21 @@ export class AddProjectDialogComponent implements OnInit {
   public smellFilters: SmellFilter[] = [];
   public metricsForSelection: string[] = [];
   public selectedSmell: string = '';
+  public numOfInstancesTypes: NumOfInstancesType[] = [NumOfInstancesType.Percentage, NumOfInstancesType.Number];
+  public projectBuildSettings: ProjectBuildSettings = new ProjectBuildSettings({numOfInstances: 100, numOfInstancesType: NumOfInstancesType.Percentage, randomizeClassSelection: true, randomizeMemberSelection: true});
+
+  public nameFormControl: FormControl = new FormControl('', [
```

Comment:
See if this heading can help you improve this code https://angular.io/guide/reactive-forms#grouping-form-controls